pkg/kubectl/cmd/testing/fake.go

```diff
@@ -680,6 +685,7 @@ func testDynamicResources() []*discovery.APIGroupResources {
 			VersionedResources: map[string][]metav1.APIResource{
 				"v1beta1": {
 					{Name: "deployments", Namespaced: true, Kind: "Deployment"},
+					{Name: "thirdpartyresources", Namespaced: false, Kind: "ThirdPartyResource"},
```

Comment:
I don't think more coverage of CRUD of known objects is needed. The gap we have is in testing unknown objects (e.g. third party resource instance objects)